pkg/lightning/backend/local/local.go

```diff
@@ -451,6 +451,8 @@ type BackendConfig struct {
 	DupeDetectEnabled  bool
 	DuplicateDetectOpt common.DupDetectOpt
 	TiKVWorkerURL      string
+	// When using global sort with DupeDetectEnabled, we can use NoopKeyAdapter instead of DupDetectKeyAdapter
+	KeyAdapter common.KeyAdapter
```

Comment:
previous we don't need it to do duplicate detect, why add it now

Reply:
This field is initialized by `genConfig` in `pkg/ddl/ingest/config.go`, and used by `newEngineManager` in `pkg/lightning/backend/local/engine_mgr.go`.
Because we can not get the caller (ADD INDEX or IMPORT INTO) and scenario (enable GLOBAL SORT or not) in `newEngineManager`.